config/v1/types_infrastructure.go

```diff
@@ -182,6 +182,12 @@ type BareMetalPlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// UseMDNS indicates where baremetal mDNS services should be deployed.
+	// See the documentation of the corresponding install-config member at
+	// https://github.com/openshift/installer/blob/master/pkg/types/baremetal/platform.go
+	// for details on which values are recognized.
+	UseMDNS string `json:"useMDNS,omitempty"`
```

Comment:
This does not feel like the right place for this.  I think this is instead of a configuration option on the dns config of a cluster.  There is nothing specific to bare metal about this option.

Reply:
@cybertron would it better to add "UseMDNS" to other metal3 config items? See https://github.com/openshift/api/pull/540 for more details. It is still very much WIP.

Reply:
Possibly. It just needs to be consumable in the machine-config-operator templates.

I will note that this _is_ baremetal-specific. It's used in the same MCO templates as the other entries in this structure, and it won't be used anywhere else. Most platforms don't use mdns services in the first place, and the ones that do have their own platform-specific configuration for it.

Reply:
The machine-api-operator would also have to look at this configuration before deploying metal3.